TrafficSimulator/Assets/Prefabs/Car/AutoDrive.cs

```diff
@@ -159,16 +169,55 @@ void Update()
             {
                 P_MoveToNextPosition();
             }
+            updateOccupiedNodes();
+        }
+
+        // Update the list of nodes that the vehicle is currently occupying
+        private void updateOccupiedNodes()
+        {
+            foreach (LaneNode node in _occupiedNodes)
+            {
+                node.UnsetVehicle(_vehicle);
+            }
+            _occupiedNodes = getOccupiedNodes();
+            foreach (LaneNode node in _occupiedNodes)
+            {
+                node.SetVehicle(_vehicle);
+            }
+        }
+
+        // Get the list of nodes that the vehicle is currently occupying by moving backwards from the current position until out of vehicle bounds
+        private List<LaneNode> getOccupiedNodes()
+        {
+            List<LaneNode> nodes = new List<LaneNode>();
+            LaneNode node = _currentNode;
+            while (node != null && Vector3.Distance(node.Position, transform.position) <= _vehicleLength/1.5)
+            {
+                nodes.Add(node);
+                node = node.Prev;
+            }
+            try {
+                node = _currentNode.Next;
+            } catch (Exception e) {
+                Debug.Log(e);
+            }
+            while (node != null && Vector3.Distance(node.Position, transform.position) <= _vehicleLength/1.5)
+            {
+                nodes.Add(node);
+                node = node.Next;
+            }
+            return nodes;
         }
 
         private void Q_TeleportToLane()
         {
             // Move it to the current position, offset in the opposite direction of the lane
-            transform.position = _currentNode.Position - (2 * (_currentNode.Rotation * Vector3.forward).normalized);
+            transform.position = _currentNode.Position - (_currentNode.Rotation * Vector3.forward * _vehicleLength / 2);
```

Comment:
Offset should not be dependent on vehicle length as it is only used to make sure the vehicle is behind the starting node. Revert this change